core/src/main/java/smile/classification/OneVersusRest.java

```diff
@@ -160,7 +218,7 @@ public int predict(T x, double[] posteriori) {
         }
 
         for (int i = 0; i < k; i++) {
-            posteriori[i] = platts[i].scale(classifiers[i].f(x));
+            posteriori[i] = platts != null? platts[i].scale(classifiers[i].f(x)) : classifiers[i].predict(x);
```

Comment:
same problem would arise here (if there was no error thrown as you added in master).
indeed the posteriori array would be an array of zeros and ones. the whichMax utility function will choose the first one, so the predictions will be the same as above.